recipe/activate.sh

```diff
@@ -17,6 +17,27 @@ export CC_@CONDA_RUST_HOST_LOWER@="${CC_FOR_BUILD:-${CONDA_PREFIX}/bin/@rust_def
 
 if [[ "@cross_target_platform@" == linux*  ]]; then
   export CARGO_BUILD_RUSTFLAGS="-C link-arg=-Wl,-rpath-link,${PREFIX:-${CONDA_PREFIX}}/lib -C link-arg=-Wl,-rpath,${PREFIX:-${CONDA_PREFIX}}/lib"
+elif [[ "@cross_target_platform@" == win* ]]; then
+  export CARGO_TARGET_@rust_arch_env@_LINKER=${CONDA_PREFIX}/bin/lld-link
+
+  # some rust crates need a linux gnu c compiler at buildtime
+  # thus we need to create custom cflags since the default ones are for clang
+  export AR_@CONDA_RUST_HOST_LOWER@="${AR}"
+  export AR_@CONDA_RUST_TARGET_LOWER@=$CONDA_PREFIX/bin/llvm-lib
+
+  export CFLAGS_@CONDA_RUST_HOST_LOWER@=""
+  export CFLAGS_@CONDA_RUST_TARGET_LOWER@="${CFLAGS}"
+
+  export CPPFLAGS_@CONDA_RUST_TARGET_LOWER@="${CPPFLAGS}"
+  export CPPFLAGS_@CONDA_RUST_HOST_LOWER@=""
+
+  export CC_@CONDA_RUST_TARGET_LOWER@=$CONDA_PREFIX/bin/clang-cl
+  export CXX_@CONDA_RUST_TARGET_LOWER@=$CONDA_PREFIX/bin/clang-cl
```

Comment:
We don't seem to set the CXX_* related variables elsewhere. Should we be setting `CXX_@CONDA_RUST_HOST_LOWER@` and `CPPFLAGS_@CONDA_RUST_???_LOWER@`?

Reply:
The problem occurs here only because the gcc toolchain tries to use our clang flags. Normally this is not the case because we do not mix C compilers. Settings `CFLAGS_@CONDA_RUST_HOST_LOWER@=""` also should clear any flags, i am not sure if clang would continue to work.

I think we should generally store the CFLAGS of gcc and clang additionally in special variables CFLAGS_gcc and CFLAGS_clang, because currently they overwrite each other on activation and the original CFLAGS of gcc are lost

Reply:
We don't have that mixup with CXX?

Reply:
We are setting `CC_@CONDA_RUST_TARGET_LOWER@`, `CXX_@CONDA_RUST_TARGET_LOWER@`, `CFLAGS_@CONDA_RUST_HOST_LOWER@` and `CPPFLAGS_@CONDA_RUST_TARGET_LOWER@`. Which variable would you set additionally?

Reply:
We are now also setting the CXXFLAGS, i believed CPPFLAGS are the CPlusPlus flags